src/cudadev/CUDACore/ScopedContext.cc

```diff
@@ -81,8 +54,22 @@ namespace cms::cuda {
     }
 
     void ScopedContextHolderHelper::enqueueCallback(int device, cudaStream_t stream) {
-      cudaCheck(
-          cudaStreamAddCallback(stream, cudaScopedContextCallback, new CallbackData{waitingTaskHolder_, device}, 0));
+      SharedEventPtr event = getEventCache().get();
+      cudaEventRecord(event.get(), stream);
+      edm::async(waitingTaskHolder_, [device, stream, event = std::move(event)]() mutable {
```

Comment:
Thanks for point it out, this indeed could use `std::move()` (both of the calls to `enqueueCallback()` come from destructors of classes with `ScopedContextHolderHelper` member, and therefore nothing will use the the `waitingTaskHolder_`).

Copying is technically tine too (just a little bit less efficient) because indeed necessary internals are shared among copies and reference counted. The framework also holds one copy with a lifetime a tiny bit longer than the `EDModule::acquire()` call; to guarantee that the `produce()` can not be called too early.